src/datafusion/table_factory.rs

```diff
@@ -239,4 +307,122 @@ mod tests {
 
         Ok(())
     }
+
+    #[tokio::test]
+    async fn test_partitions() -> Result<(), Box<dyn std::error::Error>> {
+        let mut state = SessionState::new_with_config_rt(
+            SessionConfig::default(),
+            Arc::new(RuntimeEnv::default()),
+        );
+
+        state
+            .table_factories_mut()
+            .insert("ZARR".into(), Arc::new(super::ZarrListingTableFactory {}));
+
+        let test_data = get_test_v2_data_path("lat_lon_w_groups_example.zarr".to_string());
+
+        let sql = format!(
+            "CREATE EXTERNAL TABLE zarr_table (
```

Comment:
Not for this PR, but adding slt tests might be nice at some point. DataFusion uses to remove some of this SQL.

Reply:
hmm what's an slt test here?

Reply:
Sorry, should've included a link... https://github.com/risinglightdb/sqllogictest-rs

Reply:
Huh interesting... yeah probably worth looking into at some point.